mobile/src/user-preferences-screen/user-preferences-screen.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { Component } from 'react';
+import { RefreshControl, SafeAreaView, ScrollView } from 'react-native';
+import { connect } from 'react-redux';
+import { loadUserPreferences, updateUserPreferences, UserActions } from '../reducers/user/user.action';
+import { UserPreferences } from '../reducers/user/user-preferences.model';
+import { AppState } from '../reducers/app.reducer';
+import { LoadingView } from '../navigation/loading';
+import { preferencesStyles } from './preferences.styles';
+import { SwitchSettingsView } from './switch-setting-view';
+import { NavigationScreenConfig, NavigationStackScreenOptions } from 'react-navigation';
+import { Dispatch } from 'redux';
+import { Nullable } from 'types';
+
+interface UserPreferencesScreenProps {
+    userId: Nullable<string>;
+    preferences: Nullable<UserPreferences>;
+}
+
+const mapStateToProps = (state: AppState): UserPreferencesScreenProps => ({
+    userId: state.userInfo ? state.userInfo.employeeId : null,
+    preferences: state.userInfo ? state.userInfo.preferences : null,
+});
+
+interface UserPreferencesDispatchProps {
+    loadUserPreferences: (userId: string) => void;
+    updateUserPreferences: (userId: string, previousPreferences: UserPreferences, newPreferences: UserPreferences) => void;
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<UserActions>): UserPreferencesDispatchProps => ({
+    loadUserPreferences: (userId: string) => {
+        dispatch(loadUserPreferences(userId));
+    },
+    updateUserPreferences: (userId: string, previousPreferences: UserPreferences, newPreferences: UserPreferences) => {
+        dispatch(updateUserPreferences(userId, previousPreferences, newPreferences));
+    }
+});
+
+class UserPreferencesScreenImpl extends Component<UserPreferencesScreenProps & UserPreferencesDispatchProps> {
+
+    public static navigationOptions: NavigationScreenConfig<NavigationStackScreenOptions> = {
+        headerTitle: 'Preferences',
+        headerTitleStyle: preferencesStyles.headerTitle,
+    };
+
+    public componentDidMount() {
+        this.loadPreferences();
```

Comment:
I'm not sure about loading preferences once we open this view. It might be better to load them after user has been loaded and combining them into one model (instead USER_FINISHED + USER_PREFERENCES_FINISHED). Two possible reasons: a) less actions b) some other screens might depend on this model, you don't want this thing to appear in store only after you visited preferences window. The action might stay for 'reload' purpose though.

Reply:
a4256bedc3b14e82fbd8e9af6d8dbb7083679bc8 Now user preferences are loaded with user employee details just after user loading is finished. Logout happens in case of error.

Reply:
what's the point in having 'update' and 'load' actions then? There are three /user-preferences/ calls in the code right now, that should not happen

Reply:
I mean, there are two ways:
1) LOAD(/REFRESH)_USER (calls to user / user-preferences) combined into one model, recall this method on refresh
2) LOAD_USER -> LOAD_USER_FINISHED -> LOAD_USER_PREFERENCES -> LOAD_USER_PREFERENCES_FINISHED, recall this action on refresh.

Reply:
@santee do we aware of preferences being changed before entering Preferences screen?

Reply:
done 66f53d8a1641ee2faf8130b69e7526f92d6a75fa